File: Week 5/1. Service/ros2_ws/src/pkg_py/pkg_py/add_two_ints_server.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts

class AddTwoIntegers(Node):
    def __init__(self):
        super().__init__("add_two_integers")
        self.server_ = self.create_service(
            AddTwoInts, "add_two_ints", self.add_two_ints_callback)
        self.get_logger().info("Server is ON waiting for a client...")

    def add_two_ints_callback(self, request, response):
        response.sum = request.a + request.b
        self.get_logger().info(f'{request.a} + {request.b} = {response.sum}')
        return response

def main(args=None):
    rclpy.init(args=args)
    summation = AddTwoIntegers()
    rclpy.spin(summation)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: Week 5/1. Service/ros2_ws/src/pkg_cpp_py/src/add_two_ints_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

void add(const example_interfaces::srv::AddTwoInts::Request::SharedPtr request, example_interfaces::srv::AddTwoInts::Response::SharedPtr response)
{
    response->sum = request->a + request->b;
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<rclcpp::Node>("add_two_ints_server");

    rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr service =
        node->create_service<example_interfaces::srv::AddTwoInts>("add_two_ints", &add);

    RCLCPP_INFO(node->get_logger(), "Ready to add two ints.");

    rclcpp::spin(node);
    rclcpp::shutdown();
}
